Library/Config.ts

```diff
@@ -45,16 +47,28 @@ class Config {
             "*.blob.core.usgovcloudapi.net"];
         
         this.correlationId = CorrelationIdManager.correlationIdPrefix; // Initialize with a blank correlation ID until we fetch the correct value.
-        // Async to allow caller to set profileQueryEndpoint if they wish
-        setTimeout(() =>
-            CorrelationIdManager.queryCorrelationId(
-                this.profileQueryEndpoint,
-                this.instrumentationKey,
-                this.correlationIdRetryInterval,
-                (correlationId) => this.correlationId = correlationId),
-            0);
+
+        this.setCorrelationId = (correlationId) => this.correlationId = correlationId;
+
+        CorrelationIdManager.queryCorrelationId(
+            this.profileQueryEndpoint,
+            this.instrumentationKey,
+            this.correlationIdRetryInterval,
+            this.setCorrelationId);
     }
 
+    public setProfileQueryEndpoint(endpoint: string) {
```

Comment:
I think the get/set syntax for accessors in TS would be preferable (see https://www.typescriptlang.org/docs/handbook/classes.html)/ Then `this.profileQueryEndpoint` can be set directly and still trigger the appid query